gridstatus/ercot.py

```diff
@@ -576,6 +580,123 @@ def get_spp(
             )
         return Ercot._finalize_spp_df(df, settlement_point_field, locations)
 
+    @support_date_range(frequency="1Y", update_dates=ercot_update_dates)
+    def get_as_prices(
+        self,
+        date,
+        end=None,
+        verbose=False,
+    ):
+        """Get ancillary service clearing prices in hourly intervals in Day Ahead Market
+
+        Arguments:
+            date (datetime.date, str): date of delivery for AS services
+
+            end (datetime.date, str, optional): if declared, function will return
+                data as a range, from "date" to "end"
+
+            verbose (bool, optional): print verbose output. Defaults to False.
+
+        Returns:
+
+            pandas.DataFrame: A DataFrame with prices for "Non-Spinning Reserves", \
+                "Regulation Up", "Regulation Down", "Responsive Reserves".
+
+        Source:
+            https://www.ercot.com/mp/data-products/data-product-details?id=NP4-181-ER
+        """
+
+        # use to check if we need to pull daily files
+        if (
+            not end
+            and date.date()
+            >= (
+                pd.Timestamp.now(tz=self.default_timezone)
+                - pd.DateOffset(days=self.AS_PRICES_HISTORICAL_MAX_DAYS)
+            ).date()
+        ):
+            return self._get_as_prices_recent(date)
+        elif not end:
+            end = date
+
+        print(date, end)
+
+        doc_info = self._get_document(
+            report_type_id=HISTORICAL_DAM_CLEARING_PRICES_FOR_CAPACITY_RTID,
+            constructed_name_contains=f"{date.year}.zip",
+            verbose=verbose,
+        )
+
+        x = utils.get_zip_file(doc_info.url)
+        data = pd.read_csv(x)
+
+        data["Time"] = pd.to_datetime(
```

Comment:
most of this logic looks copy and pasted from the existing get_as_price. we should find away to not have to repeat it.

Reply:
refactored shared logic into `_finalize_as_price_df`